flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/StackTraceSampleServiceTest.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.flink.runtime.taskexecutor;
+
+import org.apache.flink.api.common.time.Time;
+import org.apache.flink.runtime.concurrent.ScheduledExecutor;
+import org.apache.flink.runtime.concurrent.ScheduledExecutorServiceAdapter;
+import org.apache.flink.runtime.executiongraph.ExecutionAttemptID;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import static org.hamcrest.Matchers.arrayWithSize;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.lessThan;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+
+/**
+ * Tests for {@link StackTraceSampleService}.
+ */
+public class StackTraceSampleServiceTest {
+
+	private ScheduledExecutorService scheduledExecutorService;
+
+	private StackTraceSampleService stackTraceSampleService;
+
+	@Before
+	public void setUp() throws Exception {
+		scheduledExecutorService = Executors.newSingleThreadScheduledExecutor();
+		final ScheduledExecutor scheduledExecutor = new ScheduledExecutorServiceAdapter(scheduledExecutorService);
+
+		stackTraceSampleService = new StackTraceSampleService(scheduledExecutor);
+	}
+
+	@After
+	public void tearDown() throws Exception {
+		if (scheduledExecutorService != null) {
+			scheduledExecutorService.shutdownNow();
+			scheduledExecutorService.awaitTermination(10, TimeUnit.SECONDS);
```

Comment:
We could use `ExecutorUtils#gracefulShutdown` here.

Reply:
Will fix before merging.